google/cloud/workstations/v1/workstations_connection_idempotency_policy.cc

```diff
@@ -0,0 +1,148 @@
+  return Idempotency::kNonIdempotent;
+}
+
+Idempotency WorkstationsConnectionIdempotencyPolicy::GetWorkstationConfig(
+    google::cloud::workstations::v1::GetWorkstationConfigRequest const&) {
+  return Idempotency::kIdempotent;
+}
+
+Idempotency WorkstationsConnectionIdempotencyPolicy::ListWorkstationConfigs(
+    google::cloud::workstations::v1::ListWorkstationConfigsRequest) {  // NOLINT
+  return Idempotency::kIdempotent;
+}
+
+Idempotency
+WorkstationsConnectionIdempotencyPolicy::ListUsableWorkstationConfigs(
+    google::cloud::workstations::v1::
+        ListUsableWorkstationConfigsRequest) {  // NOLINT
+  return Idempotency::kIdempotent;
+}
+
+Idempotency WorkstationsConnectionIdempotencyPolicy::CreateWorkstationConfig(
+    google::cloud::workstations::v1::CreateWorkstationConfigRequest const&) {
+  return Idempotency::kNonIdempotent;
+}
+
+Idempotency WorkstationsConnectionIdempotencyPolicy::UpdateWorkstationConfig(
+    google::cloud::workstations::v1::UpdateWorkstationConfigRequest const&) {
+  return Idempotency::kNonIdempotent;
+}
+
+Idempotency WorkstationsConnectionIdempotencyPolicy::DeleteWorkstationConfig(
+    google::cloud::workstations::v1::DeleteWorkstationConfigRequest const&) {
+  return Idempotency::kNonIdempotent;
+}
+
+Idempotency WorkstationsConnectionIdempotencyPolicy::GetWorkstation(
+    google::cloud::workstations::v1::GetWorkstationRequest const&) {
+  return Idempotency::kIdempotent;
+}
+
+Idempotency WorkstationsConnectionIdempotencyPolicy::ListWorkstations(
+    google::cloud::workstations::v1::ListWorkstationsRequest) {  // NOLINT
+  return Idempotency::kIdempotent;
+}
+
+Idempotency WorkstationsConnectionIdempotencyPolicy::ListUsableWorkstations(
+    google::cloud::workstations::v1::ListUsableWorkstationsRequest) {  // NOLINT
+  return Idempotency::kIdempotent;
+}
+
+Idempotency WorkstationsConnectionIdempotencyPolicy::CreateWorkstation(
+    google::cloud::workstations::v1::CreateWorkstationRequest const&) {
+  return Idempotency::kNonIdempotent;
+}
+
+Idempotency WorkstationsConnectionIdempotencyPolicy::UpdateWorkstation(
+    google::cloud::workstations::v1::UpdateWorkstationRequest const&) {
+  return Idempotency::kNonIdempotent;
+}
+
+Idempotency WorkstationsConnectionIdempotencyPolicy::DeleteWorkstation(
+    google::cloud::workstations::v1::DeleteWorkstationRequest const&) {
+  return Idempotency::kNonIdempotent;
+}
+
+Idempotency WorkstationsConnectionIdempotencyPolicy::StartWorkstation(
+    google::cloud::workstations::v1::StartWorkstationRequest const&) {
+  return Idempotency::kNonIdempotent;
+}
+
+Idempotency WorkstationsConnectionIdempotencyPolicy::StopWorkstation(
+    google::cloud::workstations::v1::StopWorkstationRequest const&) {
+  return Idempotency::kNonIdempotent;
+}
+
+Idempotency WorkstationsConnectionIdempotencyPolicy::GenerateAccessToken(
+    google::cloud::workstations::v1::GenerateAccessTokenRequest const&) {
+  return Idempotency::kNonIdempotent;
```

Comment:
this one could be idempotent. https://github.com/googleapis/googleapis/blob/f0cec0b201737db7e99ff1bc6dac09e16ed2ce65/google/cloud/workstations/v1/workstations_grpc_service_config.json#L13

"I don't want to micromanage these things" is also a reasonable response.

Reply:
I don't think we should even try to special-case things, except in egregious (and probably customer-raised) instances.  So, I'll leave it alone for now.  Thanks.